tools/lint/generic_checks.py

```diff
@@ -35,8 +35,11 @@ def check_trailing_whitespace(file_paths):
     for path in file_paths:
         try:
             with open(path, "r", encoding="utf-8") as f:
-                lines = files.read().splitlines()
-                error_lines = [idx if line.endswith(" ") for idx, line in enumerate(lines, start=1)]
+                lines = f.readlines()
+                error_lines = []
+                for idx, line in enumerate(lines, start=1):
+                    if line.rstrip("\n") != line.rstrip():
+                        error_lines.append(idx)
```

Comment:
The for and if had switched lines in the previously suggested list comprehension:

```suggestion
                lines = files.read().splitlines()
                error_lines = [
                    idx
                    for idx, line in enumerate(lines, start=1)
                    if line.endswith(" ")
                ]
```

In your version `rstrip("\n")` doesn't do anything as `splitlines()` removes newlines:
> Line breaks are not included in the resulting list unless `keepends` is given and true.

However the suggestion above does not check for all types of white space like your original code was. To do that we could rewrite it to:

```suggestion
                lines = files.read().splitlines()
                error_lines = [
                    idx
                    for idx, line in enumerate(lines, start=1)
                    if line != line.rstrip()
                ]
```

Note: This suggestion is not tested either.